app/assets/javascripts/components/autocomplete.js

```diff
@@ -4,18 +4,63 @@ var $autocompletes = document.querySelectorAll('[data-module="autocomplete"]')
 if ($autocompletes) {
   $autocompletes.forEach(function ($el) {
     var customAttributes = {}
-    if ($el.attributes['data-contextual-guidance']) {
+    var $select = $el.querySelector('select')
+    if ($select.attributes['data-contextual-guidance']) {
       customAttributes = {
-        'data-contextual-guidance': $el.attributes['data-contextual-guidance'].value
+        'data-contextual-guidance': $select.attributes['data-contextual-guidance'].value
       }
     }
 
     // disabled eslint because we can not control the name of the constructor (expected to be EnhanceSelectElement)
     new window.accessibleAutocomplete.enhanceSelectElement({ // eslint-disable-line no-new, new-cap
-      selectElement: $el,
+      selectElement: $select,
       minLength: 3,
       showNoOptionsFound: true,
       customAttributes: customAttributes
     })
   })
 }
+
+var $customTemplateAutocomplete = document.querySelector('[data-module="autocomplete-custom-template"]')
+if ($customTemplateAutocomplete) {
+  // Read options and associated data attributes and feed that as results for inputValueTemplate
+  var $select = $customTemplateAutocomplete.querySelector('select')
+  var $options = $select.querySelectorAll('option')
+
+  // Create wrapper to inject the autocomplete element
+  $customTemplateAutocomplete.insertAdjacentHTML('beforeend', '<div class="autocomplete__wrapper"></div>')
+
+  // Remove select element from DOM
+  $select.remove()
+
+  new window.accessibleAutocomplete({ // eslint-disable-line no-new, new-cap
+    element: $customTemplateAutocomplete.querySelector('.autocomplete__wrapper'),
+    id: $select.id,
+    source: function (query, syncResults) {
+      var results = []
+      $options.forEach(function ($el) {
+        results.push({value: $el.textContent, hint: $el.dataset.hint || ''})
+      })
+
+      syncResults(query
+        ? results.filter(function (result) {
```

Comment:
Minor but I think there's probably a bit too much code here for a ternary operator